www/src/views/XMultiCredentialView.tsx

```diff
@@ -0,0 +1,151 @@
+import { useMutation, useQuery } from "@apollo/react-hooks";
+import gql from "graphql-tag";
+import * as React from "react";
+import { FunctionComponent } from "react";
+import { Button, Table } from "semantic-ui-react";
+import { XNoCredentialsFound } from "../components/credential";
+import { XBoundary } from "../components/layout";
+import { XErrorMessage, XLoadingMessage } from "../components/messages";
+import { Credential } from "../graphql/models";
+
+export const MULTI_CREDENTIAL_QUERY = gql`
+  {
+    credentials {
+      id
+      principal
+      secret
+      kind
+      fails
+      target {
+        id
+        name
+      }
+    }
+  }
+`;
+
+const REMOVE_CREDENTIAL = gql`
+  mutation RemoveCredential($credential: ID!) {
+    deleteCredential(input: { id: $credential })
+  }
+`;
+
+type RemoveCredentialButtonProps = {
+  credential: Credential;
+};
+
+const RemoveCredentialButton: FunctionComponent<RemoveCredentialButtonProps> = ({
+  credential
+}) => {
+  const [removeCredential] = useMutation(REMOVE_CREDENTIAL, {
+    refetchQueries: [{ query: MULTI_CREDENTIAL_QUERY }]
+  });
+
+  const handleRemove = () => {
+    let vars = {
+      credential: credential.id
+    };
+
+    removeCredential({
+      variables: vars
+    });
+  };
+
+  return (
+    <Button.Group labeled>
+      <Button compact negative icon="x" onClick={handleRemove} />
+    </Button.Group>
+  );
+};
+
+export type MultiCredentialResponse = {
+  credentials: Credential[];
+};
+
+type XCredentialTableRowProps = {
+  credential: Credential;
+};
+
+const XCredentialTableRow: FunctionComponent<XCredentialTableRowProps> = ({
+  credential
+}) => {
+  const whenEmpty = <span>No Credentials</span>;
+
+  return (
+    <Table.Row>
+      <Table.Cell width={2}>
+        <b>{credential.principal}</b>
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        <b>{credential.secret}</b>
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        {credential.kind}
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        {credential.target.name}
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        <RemoveCredentialButton credential={credential} />
+      </Table.Cell>
+    </Table.Row>
+  );
+};
+
+const XMultiCredentialView = () => {
+  const { loading, error, data: { credentials = [] } = {} } = useQuery<
+    MultiCredentialResponse
+  >(MULTI_CREDENTIAL_QUERY, {
+    pollInterval: 5000
+  });
+
+  const whenLoading = (
+    <XLoadingMessage
+      title="Loading Credentials"
+      msg="Fetching Credential info"
+    />
+  );
+  const whenEmpty = (
+    <Table.Row>
+      <Table.HeaderCell colSpan="5">
+        <XNoCredentialsFound />
+      </Table.HeaderCell>
+    </Table.Row>
+  );
+
+  return (
+    <React.Fragment>
+      <XErrorMessage title="Error Loading Credentials" err={error} />
```

Comment:
nit: With the new `XView` changes, your view with receive a `setError` prop that you can just set. We're trying to get rid of the need for `XBoundary`.

Reply:
I do not understand this. I do not know how this would work or how to change to address this.